terraform/eval_interpolate.go

```diff
@@ -22,3 +26,28 @@ func (n *EvalInterpolate) Eval(ctx EvalContext) (interface{}, error) {
 
 	return nil, nil
 }
+
+// EvalTryInterpolate is an EvalNode implementation that takes a raw
+// configuration and interpolates it, but only logs a warning on an
+// interpolation error, and stops further Eval steps.
+// This is used during Input where a value may not be known before Refresh, but
+// we don't want to block Input.
+type EvalTryInterpolate struct {
+	Config   *config.RawConfig
+	Resource *Resource
+	Output   **ResourceConfig
+}
+
+func (n *EvalTryInterpolate) Eval(ctx EvalContext) (interface{}, error) {
+	rc, err := ctx.Interpolate(n.Config, n.Resource)
+	if err != nil {
```

Comment:
I think another possible outcome here is that `err` is nil but `tc.ComputedKeys` is non-empty. In that case the interpolation succeeded but the returned config is incomplete. Do you think we should treat that as an interpolation failure here too? I think otherwise we'd get downstream errors e.g. when a provider tries to use an interpolation expression as a hostname, or similar.